src/librustdoc/html/url_parts_builder.rs

```diff
@@ -105,9 +105,7 @@ impl UrlPartsBuilder {
 /// This is just a guess at the average length of a URL part,
 /// used for [`String::with_capacity`] calls in the [`FromIterator`]
 /// and [`Extend`] impls.
-///
-/// This is intentionally on the lower end to avoid overallocating.
-const AVG_PART_LENGTH: usize = 5;
+const AVG_PART_LENGTH: usize = 8;
```

Comment:
Yes, jemalloc's size classes are as follows: 8, 16, 32, 48, 64, 80, 96, 12, 128, 160, 192, 224, 256, etc., so 8 is a good choice here. Might be worth adding something like "and jemalloc's size classes are all multiples of 8."  ?

Reply:
Sure, I can do that.